src/Application/Revisions/EventHandlers/RevisionCreatedEventHandler.cs

```diff
@@ -39,11 +39,12 @@ public async Task Handle(DomainEventNotification<CreatedEvent<Revision>> notific
             if (channel.RevisionSelectionStrategy == ChannelRevisionSelectionStrategy.UseRangeRule)
             {
                 var activeRevision = RevisionRangeRule.Parse(channel.RangeRule).Match(channel.App.Revisions);
-                if (activeRevision is not null && activeRevision != channel.ActiveRevision)
+                if (activeRevision is not null && activeRevision != channel.ActiveRevision && activeRevision.RevisionNumber == domainEvent.Entity.RevisionNumber)
```

Comment:
This is to prevent an issue that can happen when more revisions are imported at the same time. If you import 2 or more revisions, they are first saved to the database and then one event triggers for each one of them. Let's consider we import `spin-hello-world` versions 1.0.0 and 1.0.1. At line 41 the active revision is calculated it applied the RevisionRangeRule to the channel revisions  which will always return 1.0.1. This will actually mean that when the 1.0.0 revision is imported it will deploy the 1.0.1 revision. The check is there to make sure that the event only handles the deploy of the revision it is handling

Reply:
Ok, so this handles the case where two events are being fired simultaneously, both of which would trigger a deploy. Gotcha.